src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -2198,6 +2200,125 @@ static inline __m128 atan2_ps( __m128 y, __m128 x )
+
+static inline __m256 atan_ps( __m256 x )
+{
+    __m256 sign_bit, y;
+
+    sign_bit = x;
+    // Take the absolute value
+    x = _mm256_and_ps( x, _ps_inv_sign_mask_avx );
+    // Extract the sign bit (upper one)
+    sign_bit = _mm256_and_ps( sign_bit, _ps_sign_mask_avx );
+
+    // Range reduction, init x and y depending on range
+
+    // x > 2.414213562373095
+    __m256 cmp0 = _mm256_cmp_ps( x, _ps_atanrange_hi_avx, _CMP_GT_OS );
+    // x > 0.4142135623730950
+    __m256 cmp1 = _mm256_cmp_ps( x, _ps_atanrange_lo_avx, _CMP_GT_OS );
+
+    // x > 0.4142135623730950 && !( x > 2.414213562373095 )
+    __m256 cmp2 = _mm256_andnot_ps( cmp0, cmp1 );
+
+    // -( 1.0/x )
+    __m256 y0 = _mm256_and_ps( cmp0, _ps_cephes_PIO2F_avx );
+    __m256 x0 = _mm256_div_ps( _ps_1_avx, x );
+    x0 = _mm256_xor_ps( x0, _ps_sign_mask_avx );
+
+    __m256 y1 = _mm256_and_ps( cmp2, _ps_cephes_PIO4F_avx );
+    // (x-1.0)/(x+1.0)
+    __m256 x1_o = _mm256_sub_ps( x, _ps_1_avx );
+    __m256 x1_u = _mm256_add_ps( x, _ps_1_avx );
+    __m256 x1 = _mm256_div_ps( x1_o, x1_u );
+
+    __m256 x2 = _mm256_and_ps( cmp2, x1 );
+    x0 = _mm256_and_ps( cmp0, x0 );
+    x2 = _mm256_or_ps( x2, x0 );
+    cmp1 = _mm256_or_ps( cmp0, cmp2 );
+    x2 = _mm256_and_ps( cmp1, x2 );
+    x = _mm256_andnot_ps( cmp1, x );
+    x = _mm256_or_ps( x2, x );
+
+    y = _mm256_or_ps( y0, y1 );
+
+    __m256 zz = _mm256_mul_ps( x, x );
+    __m256 acc = _ps_atancof_p0_avx;
+    acc = _mm256_mul_ps( acc, zz );
+    acc = _mm256_sub_ps( acc, _ps_atancof_p1_avx );
+    acc = _mm256_mul_ps( acc, zz );
+    acc = _mm256_add_ps( acc, _ps_atancof_p2_avx );
+    acc = _mm256_mul_ps( acc, zz );
+    acc = _mm256_sub_ps( acc, _ps_atancof_p3_avx );
+    acc = _mm256_mul_ps( acc, zz );
+    acc = _mm256_mul_ps( acc, x );
+    acc = _mm256_add_ps( acc, x );
+    y = _mm256_add_ps( y, acc );
+
+    // Update the sign
+    y = _mm256_xor_ps( y, sign_bit );
+
+    return y;
+}
+
+static inline __m256 atan2_ps( __m256 y, __m256 x )
+{
+    __m256 x_eq_0 = _mm256_cmp_ps( x, _ps_0_avx, _CMP_EQ_OQ );
+    __m256 x_gt_0 = _mm256_cmp_ps( x, _ps_0_avx, _CMP_GT_OS);
+    __m256 x_le_0 = _mm256_cmp_ps( x, _ps_0_avx, _CMP_LE_OS );
+    __m256 y_eq_0 = _mm256_cmp_ps( y, _ps_0_avx, _CMP_EQ_OQ );
+    __m256 x_lt_0 = _mm256_cmp_ps( x, _ps_0_avx, _CMP_LT_OS );
+    __m256 y_lt_0 = _mm256_cmp_ps( y, _ps_0_avx, _CMP_LT_OS );
+
+    __m256 zero_mask = _mm256_and_ps( x_eq_0, y_eq_0 );
+    __m256 zero_mask_other_case = _mm256_and_ps( y_eq_0, x_gt_0 );
+    zero_mask = _mm256_or_ps( zero_mask, zero_mask_other_case );
+
+    __m256 pio2_mask = _mm256_andnot_ps( y_eq_0, x_eq_0 );
+    __m256 pio2_mask_sign = _mm256_and_ps( y_lt_0, _ps_sign_mask_avx );
+    __m256 pio2_result = _ps_cephes_PIO2F_avx;
+    pio2_result = _mm256_xor_ps( pio2_result, pio2_mask_sign );
+    pio2_result = _mm256_and_ps( pio2_mask, pio2_result );
+
+    __m256 pi_mask = _mm256_and_ps( y_eq_0, x_le_0 );
+    __m256 pi = _ps_cephes_PIF_avx;
```

Comment:
Need not use another variable here for PI. Directly try to use _ps_cephes_PIF_avx